sdk/resourcemanager/Azure.ResourceManager/src/Custom/Resources/Models/Location.cs

```diff
@@ -317,141 +299,44 @@ public override string ToString()
-
-            return Name == other.Name && CanonicalName == other.CanonicalName && DisplayName == other.DisplayName;
+            return Name == other.Name;
         }
 
         /// <summary>
-        /// Compares this Location name to another Location to expose if it is greater, less or equal than this one.
+        /// Creates a string implicitly from a Location object.
         /// </summary>
-        /// <param name="other"> Location object or name as a string. </param>
-        /// <returns> -1 for less than, 0 for equals, 1 for greater than. </returns>
-        public int CompareTo(Location other)
-        {
-            if (ReferenceEquals(other, null))
-            {
-                return 1;
-            }
-
-            return string.Compare(Name, other.Name, StringComparison.InvariantCulture);
-        }
-
-        private static string NormalizationUtility(string value)
-        {
-            if (string.IsNullOrEmpty(value))
-            {
-                return value;
-            }
-
-            var sb = new StringBuilder(value.Length);
-            for (var index = 0; index < value.Length; ++index)
-            {
-                var c = value[index];
-                if (char.IsLetterOrDigit(c))
-                {
-                    sb.Append(c);
-                }
-            }
-
-            return sb.ToString().ToUpperInvariant();
-        }
-
-        private static NameType DetectNameType(string location)
-        {
-            if (Regex.IsMatch(location, CanonicalPattern))
-            {
-                return NameType.CanonicalName;
-            }
-            else if (Regex.IsMatch(location, DisplayPattern))
-            {
-                return NameType.DisplayName;
-            }
-            else
-            {
-                return NameType.Name;
-            }
-        }
-
-        private static string GetCanonicalNameFromDisplayName(string name)
-        {
-            return name.Replace(RegexWhitespace, RegexDash).ToLower(CultureInfo.InvariantCulture);
-        }
-
-        private static string GetDisplayNameFromCanonicalName(string name)
-        {
-            char[] chName = name.ToCharArray();
-            chName[0] = char.ToUpper(chName[0], CultureInfo.InvariantCulture);
-
-            for (int i = 0; i < chName.Length - 1; i++)
-            {
-                if (chName[i] == '-')
-                {
-                    chName[i] = ' ';
-                    chName[i + 1] = char.ToUpper(chName[i + 1], CultureInfo.InvariantCulture);
-                }
-            }
-
-            return new string(chName);
-        }
-
-        private static string GetDefaultNameFromCanonicalName(string name)
-        {
-            return name.Replace(RegexDash, string.Empty);
-        }
-
-        private static string GetDefaultNameFromDisplayName(string name)
+        /// <param name="other"> Location object to be assigned. </param>
+        public static implicit operator string(Location other)
```

Comment:
is there a good use case for this operator?

Reply:
I think mostly for convenience of not having to add .ToString every time we pass it through serializer, uir creation, etc.